src/test/java/teammates/ui/webapi/FeedbackSessionLogsUpdateActionTest.java

```diff
@@ -0,0 +1,145 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionLogEntryAttributes;
+import teammates.common.datatransfer.logs.FeedbackSessionLogType;
+import teammates.common.util.Const;
+import teammates.test.ThreadHelper;
+
+/**
+ * SUT: {@link FeedbackSessionLogsUpdateAction}.
+ */
+public class FeedbackSessionLogsUpdateActionTest
+        extends BaseActionTest<FeedbackSessionLogsUpdateAction> {
+
+    private static final int MIN_WINDOW_PERIOD = 2 * 1000;
+    private static final int ISOLATION_PERIOD = 5 * 1000;
+
+    @Override
+    protected String getActionUri() {
+        return Const.CronJobURIs.AUTOMATED_FEEDBACK_SESSION_LOGS_UPDATE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Override
+    @Test
+    protected void testAccessControl() {
+        // verifyOnlyAdminCanAccess();
+    }
+
+    @Override
+    @Test
+    public void testExecute() throws Exception {
+        FeedbackSessionLogsUpdateAction action = getAction();
+
+        ______TS("creating 2 logs in the same window should return only 1 log");
+
+        // create a window exceeding max size
+        for (int i = 0; i < 2; i++) {
+            mockLogsProcessor.createFeedbackSessionLog("idOfTypicalCourse1",
+                    "student1InCourse1@gmail.tmt", "First feedback session", FeedbackSessionLogType.ACCESS.getLabel());
+            ThreadHelper.waitFor(MIN_WINDOW_PERIOD/2);
+        }
+
+        action.execute();
+
+        Instant currentTime = Instant.now();
+        List<FeedbackSessionLogEntryAttributes> allLogEntries = mockLogsProcessor.getFeedbackSessionLogs(
+                "idOfTypicalCourse1", "student1InCourse1@gmail.tmt",
+                currentTime.minus(MIN_WINDOW_PERIOD/1000 + 1, ChronoUnit.SECONDS).toEpochMilli(),
+                currentTime.toEpochMilli(), null);
+        List<FeedbackSessionLogEntryAttributes> logs = logic.getFeedbackSessionLogs(
+                "idOfTypicalCourse1", null,
+                currentTime.minus(MIN_WINDOW_PERIOD/1000 + 1, ChronoUnit.SECONDS).toEpochMilli(),
+                currentTime.toEpochMilli(), null);
+
+        assertEquals(1, logs.size());
+        assertEquals(2, allLogEntries.size());
+
+        FeedbackSessionLogEntryAttributes firstLog = logs.get(0);
+        FeedbackSessionLogEntryAttributes secondLog = allLogEntries.get(1);
+        long windowSize = secondLog.getTimestamp() - firstLog.getTimestamp();
+
+        // check that the first log is taken
+        // and the window size is smaller than the min value
+        assertTrue(windowSize > 0);
+        assertTrue(windowSize < MIN_WINDOW_PERIOD);
```

Comment:
I'm not sure why testing this `windowSize` is needed. Checking that the first log is taken is already guaranteed by the first `assertEquals` after this, and checking the window size is nothing of value here, unless we are testing multiple windows.

Reply:
this is to check that the 2nd log was discarded correctly i.e the gap was larger than the min period

Reply:
If this is checking the 2nd log was discarded, then what does this line do:
```java
assertEquals(1, logs.size());
```

Reply:
the above line check that the 2nd log was discarded, line 74 and 75 verify that it was discarded **correctly**

Reply:
I still don't get it. What cases can fail here that all other parts of this test will not fail?

Reply:
lets say for some reason, these 2 logs are 3 seconds apart, but only 1 log was returned. In this case, if we dont include line 74 and 75, the test will pass which is not what we want

Reply:
> these 2 logs are 3 seconds apart, but only 1 log was returned

If the later test "creating 2 logs in different windows should return both logs" does its job, this is an unreachable scenario.